src/Network/Network/NetworkManager/NetworkManagerSecurityAdminConfiguration/NewAzNetworkManagerSecurityAdminConfigurationCommand.cs

```diff
@@ -65,8 +70,8 @@ public class NewAzNetworkManagerSecurityAdminConfigurationCommand : NetworkManag
         [Parameter(
            Mandatory = false,
            ValueFromPipelineByPropertyName = true,
-           HelpMessage = "ApplyOnNetworkIntentPolicyBasedServices. Valid values include 'None' and 'All'.")]
-        public string[] ApplyOnNetworkIntentPolicyBasedService { get; set; }
+           HelpMessage = "How security admin rules are applied to virtual networks with services using network intent policies. Valid values include 'None', 'All', and 'AllowRulesOnly'.")]
+        public NetworkIntentPolicyBasedServiceType[] ApplyOnNetworkIntentPolicyBasedService { get; set; }
```

Comment:
Array of pre-defined type parameters may not easy to use

Reply:
As far as I could tell, 'ValidateSet' only works if you have a single string as the input. By defining the enum here, I can iterate and try to parse on each value. You don't actually have to define this input as an enum type when running the command, its still just a string.